geom/alg_exact_equals.go

```diff
@@ -155,9 +155,12 @@ func (c exactEqualsComparator) lineStringsEq(ls1, ls2 LineString) bool {
 }
 
 func (c exactEqualsComparator) pointsEq(p1, p2 Point) bool {
+	if p1.IsEmpty() && p2.IsEmpty() {
+		return p1.CoordinatesType() == p2.CoordinatesType()
```

Comment:
Can two non-empty points with different coordinate types ever be considered equal? Or perhaps this is handled by `c.eq(c1, c2)`.

Reply:
Yes, that case is handled by the call to `c.eq(c1, c2)`.